src/build/arg/arg_value.rs

```diff
@@ -18,6 +39,12 @@ impl Display for ArgValue<'_> {
     }
 }
 
+impl<'help> From<&ArgValue<'help>> for ArgValue<'help> {
+    fn from(s: &ArgValue<'help>) -> Self {
+        s.clone()
+    }
+}
+
```

Comment:
This is not a clean solution IMO but not sure how to solve this better, the issue is this usecase:
https://github.com/clap-rs/clap/blob/90996399daf6536b270ed8a3d9994a9855f23f16/src/build/arg/mod.rs#L1881-L1889

Reply:
Is the `&` at the start needed?

Reply:
not here, but it won't work with the ArgEnum trait, because we have a `&[]` there for VARIANTS.
 We probably could make the ArgEnum trait generic other a VARIANTS impl `IntoIter<...>` not sure if that is the right way forward tho.
 
 The error in #2762 :
 ```
 error[E0277]: the trait bound `ArgValue<'_>: From<&ArgValue<'_>>` is not satisfied
  --> clap_derive/examples/value_hints_derive.rs:42:23
   |
42 |     generator: Option<GeneratorChoice>,
   |                       ^^^^^^^^^^^^^^^ the trait `From<&ArgValue<'_>>` is not implemented for`ArgValue<'_>`
   |
   = help: the following implementations were found:
             <ArgValue<'help> as From<&&'help str>>
             <ArgValue<'help> as From<&'help str>>
   = note: required because of the requirements on the impl of `Into<ArgValue<'_>>` for `&ArgValue<'_>`
   ```